### PR TITLE
addded throwing Exception when the Google Api returns error

### PR DESCRIPTION
**Pull request by piggydoughnut**
We are using your bundle in our project and we need this functionality :) 
Thx :)

### CODE REVIEW

File: Services/GeocodeService.php [hunk 1]
```diff
@@ -81,4 +82,13 @@ public function reverseGeocodeAddress($lat, $lang, $raw = false)
 
         return $response->results[0]->formatted_address;
     }
+
+    public function returnException($response){
+        if ($response->error_message) {
+            $msg = $response->error_message;
+        } else {
+            $msg = 'There has been an eror while communicating with Google API';
```
Comment (mychalvlcek) [inline Services/GeocodeService.php]: please fix the typo ("eror") and respect coding standards

File: Services/GeocodeService.php [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <?php
 
 namespace HappyR\Google\GeocoderBundle\Services;
+use Exception;
```
Comment (Nyholm) [inline Services/GeocodeService.php]: Remove this line and use `\Exception` instead.

File: Services/GeocodeService.php [hunk 3]
```diff
@@ -46,7 +47,7 @@ public function geocodeAddress($text, $raw = false)
         $response = json_decode($response);
 
         if ($response->status != 'OK') {
-            return null;
+            return $this->returnException($response);
```
Comment (Nyholm) [inline Services/GeocodeService.php]: Remove "return"

File: Services/GeocodeService.php [hunk 4]
```diff
@@ -72,7 +73,7 @@ public function reverseGeocodeAddress($lat, $lang, $raw = false)
...
```
Comment (Nyholm) [inline Services/GeocodeService.php]: Remove "return"

File: Services/GeocodeService.php [hunk 5]
```diff
@@ -81,4 +82,14 @@ public function reverseGeocodeAddress($lat, $lang, $raw = false)
 
         return $response->results[0]->formatted_address;
     }
+
+    public function returnException($response)
```
Comment (Nyholm) [inline Services/GeocodeService.php]: we are not returning anything. Maybe rename to "HandleError"?

File: Services/GeocodeService.php [hunk 6]
```diff
@@ -81,4 +82,14 @@ public function reverseGeocodeAddress($lat, $lang, $raw = false)
...
+    {
+        if ($response->error_message) {
+            $msg = $response->error_message;
+        } else {
+            $msg = 'There has been an error while communicating with Google API';
+        }
+        throw new Exception($msg);
```
Comment (Nyholm) [inline Services/GeocodeService.php]: `throw new \Exception($msg);`

**Comment by Nyholm** [comment]
Awesome @piggydoughnut. This looks great. 
I've got some minor style comments but otherwise is looks fine.

**Comment by Nyholm** [comment]
Maybe it should be a config parameter if we want to throw exceptions or not. What do you think?

**Comment by piggydoughnut** [comment]
Hmm. I d leave it like this, in case someone wants to do something about that Exception they can catch it :)

**Comment by piggydoughnut** [comment]
Hello there:) Is everything else ok with PR? Any other changes u have in mind?

**Comment by Nyholm** [comment]
Thank you for this. 

I will tag a new version as soon as we can make Exceptions optional.

**Comment by Nyholm** [comment]
Tagged 1.1.0

**Pull request merged**